### PR TITLE
[Ascend] LoRA: adding Ascend LoRA backend with using kernels from sgl_kernel_npu

### PR DESCRIPTION
**Pull request by vlserov**
## Motivation

Integration LoRA kernels into SCLang in order to support LoRA adapters on Ascend devices.

## Modifications

Integrated LoRA backend, which used vllm kernels to perform sgmv operation instead of triton implementation.

## Accuracy Tests

| No LoRA enabled | Qwen3-32B | Qwen3-30B-A3B |
| :------------------: | :------------: | :-----------------: |
| Accuracy              | 0.848            | 0.904                   |
| Latency                | 270.254 s      | 148.327 s            |
| Throughput         | 1303.601 t/s | 1027.614 t/s        |

| LoRA enabled      | Qwen3-32B | Qwen3-30B-A3B |
| :------------------: | :------------: | :-----------------: |
| Accuracy              | 0.852            | 0.889                   |
| Latency                | 357.629 s      | 196.121 s            |
| Throughput         | 1003.687 t/s | 791.195 t/s          |

## Benchmarking and Profiling

LoRA adapter should be matched with served model.
For example, for model Qwen3-32B can be used adapters:

- flyfishxu/DeepNews-LoRA-Qwen3-32B
- nicoboss/Qwen3-32B-Uncensored-Lora

```
python3 -m sglang.launch_server --model-path Qwen3-32B
    --enable-lora \
    --lora-paths lora0=flyfishxu/DeepNews-LoRA-Qwen3-32B \
    --max-loras-per-batch 1
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/lora/backend/triton_backend.py [hunk 1]
```diff
@@ -97,7 +97,9 @@ def run_gate_up_lora(
         return lora_output
 
     def init_cuda_graph_batch_info(
-        self, cuda_graph_batch_info: LoRABatchInfo, max_bs_in_cuda_graph: int
+        self,
+        cuda_graph_batch_info: LoRABatchInfo,
+        max_bs_in_cuda_graph: int,
```
Comment (ping1jing2) [inline python/sglang/srt/lora/backend/triton_backend.py]: revert

File: python/sglang/srt/lora/lora_manager.py [hunk 2]
```diff
@@ -21,7 +21,8 @@
 import torch
 
 from sglang.srt.configs.load_config import LoadConfig
-from sglang.srt.lora.backend.base_backend import BaseLoRABackend, get_backend_from_name
+from sglang.srt.lora.backend.base_backend import BaseLoRABackend
```
Comment (ping1jing2) [inline python/sglang/srt/lora/lora_manager.py:24]:
```python
from torch_npu.contrib import transfer_to_npu  # noqa: F401
# Re-mock torch.cuda.is_available cuz transfer_to_npu mocks it to True
torch.cuda.is_available = lambda: False
```
add this code

File: python/sglang/srt/lora/lora_manager.py [hunk 3]
```diff
@@ -87,9 +88,9 @@ def __init__(
             lora_paths=lora_paths,
         )
 
-    def init_cuda_graph_batch_info(self, max_bs_in_cuda_graph: int):
+    def init_cuda_graph_batch_info(self, max_bs_in_cuda_graph: int, device):
         self.max_bs_in_cuda_graph = max_bs_in_cuda_graph
-        with torch.device("cuda"):
+        with torch.device(device):
```
Comment (ping1jing2) [inline python/sglang/srt/lora/lora_manager.py]: it's no need to change these code after you import `transfer_to_npu`

File: python/sglang/srt/model_executor/cuda_graph_runner.py [hunk 4]
```diff
@@ -299,7 +299,9 @@ def __init__(self, model_runner: ModelRunner):
             set_torch_compile_config()
 
         if self.model_runner.server_args.enable_lora:
-            self.model_runner.lora_manager.init_cuda_graph_batch_info(self.max_bs)
+            self.model_runner.lora_manager.init_cuda_graph_batch_info(
+                self.max_bs, self.device
```
Comment (ping1jing2) [inline python/sglang/srt/model_executor/cuda_graph_runner.py]: ditto

File: python/sglang/srt/lora/lora_manager.py [hunk 5]
```diff
@@ -37,9 +38,16 @@
 from sglang.srt.managers.io_struct import LoRAUpdateOutput
 from sglang.srt.model_executor.forward_batch_info import ForwardBatch
 from sglang.srt.server_args import ServerArgs
-from sglang.srt.utils import replace_submodule
+from sglang.srt.utils import is_npu, replace_submodule
 from sglang.srt.utils.hf_transformers_utils import AutoConfig
 
+if is_npu():
+    from torch_npu.contrib import transfer_to_npu  # noqa: F401
```
Comment (iforgetmyname) [inline python/sglang/srt/lora/lora_manager.py:45]: move these codes to `model_runner.py`, these codes need to be effective globally

Comment (ping1jing2) [reply python/sglang/srt/lora/lora_manager.py:45]: will do it late

**Review by ssshinigami — APPROVED** [review]

**Review by Fridge003 — APPROVED** [review]
